crates/packager/src/package/app/mod.rs

```diff
@@ -52,6 +52,77 @@ pub(crate) fn package(ctx: &Context) -> crate::Result<Vec<PathBuf>> {
 
     tracing::debug!("Copying frameworks");
     let framework_paths = copy_frameworks_to_bundle(&contents_directory, config)?;
+
+    // All dylib files and native executables should be signed manually
+    // It is highly discouraged by Apple to use the --deep codesign parameter in larger projects.
+    // https://developer.apple.com/forums/thread/129980
+    for framework_path in &framework_paths {
+        if let Some(framework_path) = framework_path.to_str() {
+            // Find all files in the current framework folder
+            let constructed_glob = format!("{}/**/*", framework_path);
+            let files = match glob::glob(&constructed_glob) {
+                Ok(res) => res,
+                Err(err) => {
+                    tracing::warn!("Could not construct a glob for {}: {}, please make sure your path contains no *", framework_path, err);
+                    tracing::warn!("This path will not be scanned for dylibs!");
+                    continue;
+                }
+            };
+
+            let files = files.filter_map(|p| match p {
+                Ok(v) => Some(v),
+                Err(err) => {
+                    tracing::warn!(
+                        "Error while globbing for Mach-O files in {}: {}",
+                        framework_path,
+                        err
+                    );
+                    None
+                }
+            });
+
+            // Filter all files for Mach-O headers. This will target all .dylib and native executable files
+            for file in files {
+                let Some(file) = file.to_str() else {
+                    tracing::warn!("Failed to convert {} to a valid UTF-8 string, this path will not be scanned for Mach-O header!", file.display());
+                    continue;
+                };
+
+                let Ok(metadata) = std::fs::metadata(file) else {
+                    tracing::warn!("Failed to get metadata for {}, this file will not be scanned for Mach-O header!", file);
+                    continue;
+                };
+
+                if !metadata.is_file() {
+                    continue;
+                }
+
+                let Ok(mut open_file) = std::fs::File::open(file) else {
+                    tracing::warn!("Failed to open {} for reading, this file will not be scanned for Mach-O header!", file);
+                    continue;
+                };
+
+                let mut buffer = [0; 4];
+                std::io::Read::read_exact(&mut open_file, &mut buffer)?;
+
+                // Mach-O magic numbers
+                let magic_numbers = [0xfeedface, 0xfeedfacf, 0xcafebabe, 0xcefaedfe, 0xcffaedfe];
```

Comment:
let's use a const here
```suggestion
                const MACH_O_MAGIC_NUMBERS: [u32; 5] = [0xfeedface, 0xfeedfacf, 0xcafebabe, 0xcefaedfe, 0xcffaedfe];
```